appintro/src/main/res/values/styles.xml

```diff
@@ -1,31 +1,64 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
 
+    <style name="AppIntroStyle" parent="Theme.AppCompat.DayNight.NoActionBar">
+
+    </style>
+
     <style name="AppIntroButtonStyleCompat" parent="Widget.AppCompat.Button.Borderless">
         <item name="android:background">@android:color/transparent</item>
+        <item name="android:minHeight">@dimen/appintro_button_minheight</item>
+        <item name="android:layout_height">wrap_content</item>
+        <item name="android:layout_width">wrap_content</item>
+        <item name="android:minWidth">@dimen/appintro_button_minwidth</item>
+        <item name="android:layout_margin">@dimen/appintro_button_margin</item>
     </style>
 
-    <style name="AppIntroDefaultPadding">
-        <item name="android:paddingTop">@dimen/appIntroDefaultPaddingTop</item>
-        <item name="android:paddingLeft">@dimen/appIntroDefaultPaddingLeft</item>
-        <item name="android:paddingRight">@dimen/appIntroDefaultPaddingRight</item>
-        <item name="android:paddingBottom">@dimen/appIntroDefaultPaddingBottom</item>
+    <style name="AppIntro2ButtonStyleCompat" parent="Widget.AppCompat.Button.Borderless">
+        <item name="android:background">@android:color/transparent</item>
+        <item name="android:layout_width">@dimen/appintro2_fab_size</item>
+        <item name="android:layout_height">@dimen/appintro2_fab_size</item>
+        <item name="android:layout_margin">@dimen/appintro2_fab_margin</item>
     </style>
 
-    <style name="AppIntroDefaultPadding.Intro1">
-        <item name="android:paddingBottom">@dimen/appIntroDefaultPaddingBottomIntro1</item>
-    </style>
+    <!--<style name="AppIntro3ButtonStyleCompat">-->
+    <!--<item name="android:layout_width">56dp</item>-->
+    <!--<item name="android:layout_height">56dp</item>-->
+    <!--This is work in progress-->
+    <!--</style>-->
 
-    <style name="AppIntroDefaultPadding.Intro2">
-        <item name="android:paddingBottom">@dimen/appIntroDefaultPaddingBottomIntro2</item>
+    <style name="AppIntroDefaultHeading" parent="TextAppearance.AppCompat.Display1">
+        <item name="android:textColor">@color/appintro_title_color</item>
+        <item name="android:textSize">@dimen/appintro_headtext_size</item>
+        <item name="android:textStyle">bold</item>
+        <item name="android:padding">@dimen/appintro_head_padding</item>
+        <item name="android:gravity">center</item>
+        <item name="android:layout_width">0dp</item>
```

Comment:
So I'm generally not a big fan of having `<item name="android:layout_...">` attributes in the Styles. The reason is that basically those layout attributes are actually contextually meaningful when the widget is used within a layout.

Moreover they make this style _harder_ to use if you want, let's say, to use a Heading in a layout where you want it to be `match_parent`.

I'll not call it a blocker, but if you've the same opinion, you can probably move back all the `layout_` properties inside the layout files.